pkg/kubectl/cmd/util/factory.go

```diff
@@ -227,6 +235,20 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 
 	clients := NewClientCache(clientConfig)
 
+	// We do this to delay loading a config until after flags have been
+	// parsed
+	var dynamicClientPool dynamic.ClientPool
+	var dynamicClientPoolOnce sync.Once
+	getDynamicClientPool := func() dynamic.ClientPool {
+		dynamicClientPoolOnce.Do(func() {
+			cfg, err := clientConfig.ClientConfig()
+			CheckErr(err)
+
+			dynamicClientPool = dynamic.NewClientPool(cfg, dynamic.LegacyAPIPathResolverFunc)
+		})
+		return dynamicClientPool
+	}
```

Comment:
Ok, in the future we may want to make the dynamic mapper do a lazy evaluation, but this is fine for now.  Can you open a follow up issue and assign it to me for that?